sklearn/utils/tests/test_utils.py

```diff
@@ -126,6 +130,28 @@ def test_pinvh_simple_complex():
     assert_almost_equal(np.dot(a, a_pinv), np.eye(3))
 
 
+def test_arpack_eigsh_initialization():
+    '''
+    Non-regression test that shows null-space computation is better with 
+    initialization of eigsh from [-1,1] instead of [0,1]
+    '''
+    random_state = check_random_state(42)
+
+    A = random_state.rand(50, 50)
+    A = np.dot(A.T, A)  # create s.p.d. matrix
+    A = graph_laplacian(A)
```

Comment:
Why calling `graph_laplacian` here? This is not guaranteed to be s.p.d. any more right? I get the following failure:

```
======================================================================
FAIL: Non-regression test that shows null-space computation is better with
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/volatile/ogrisel/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/volatile/ogrisel/code/scikit-learn/sklearn/utils/tests/test_utils.py", line 152, in test_arpack_eigsh_initialization
    assert_greater_equal(w[0], 0)
  File "/volatile/ogrisel/code/scikit-learn/sklearn/utils/testing.py", line 126, in assert_greater_equal
    assert a >= b, message
nose.proxy.AssertionError: -430.0444945505198 is not greater than or equal to 0
>>  assert -430.0444945505198 >= 0, '-430.0444945505198 is not greater than or equal to 0'
```    

The test pass when using `A = np.dot(A.T, A)` directly.